akka-typed/src/main/scala/akka/typed/ActorSystem.scala

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (C) 2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.typed
+
+import akka.event.EventStream
+import akka.actor.Scheduler
+import scala.concurrent.ExecutionContext
+import java.util.concurrent.Executor
+import scala.concurrent.duration.Duration
+import akka.actor.Extension
+import akka.actor.ExtensionId
+import akka.actor.ActorRefProvider
+import java.util.concurrent.ThreadFactory
+import akka.actor.DynamicAccess
+import akka.actor.ActorSystemImpl
+import com.typesafe.config.Config
+import akka.actor.ExtendedActorSystem
+import com.typesafe.config.ConfigFactory
+import scala.concurrent.ExecutionContextExecutor
+import scala.concurrent.Future
+
+/**
+ * An ActorSystem is home to a hierarchy of Actors. It is created using
+ * [[ActorSystem$.apply[T]*]] from a [[Props]] object that describes the root
+ * Actor of this hierarchy and which will create all other Actors beneath it.
+ * A system also implements the [[ActorRef]] type, and sending a message to
+ * the system directs that message to the root Actor.
+ */
+abstract class ActorSystem[-T](val name: String) extends ActorRef[T] { this: ScalaActorRef[T] ⇒
+
+  /**
+   * Access to the underlying (untyped) ActorSystem.
+   *
+   * FIXME: must prevent system.actorOf from untyped actors for custom guardians
+   */
+  protected val untyped: ExtendedActorSystem
```

Comment:
opened #16742 (periodic scheduling) and #16743 (registerOnTermination) and added the rest